src/Compute/Compute/Extension/CustomScript/SetAzureVMCustomScriptExtensionCommand.cs

```diff
@@ -123,7 +311,38 @@ public override void ExecuteCmdlet()
             {
                 ExecuteClientAction(() =>
                 {
-                    if (string.Equals(this.ParameterSetName, SetCustomScriptExtensionByContainerBlobsParamSetName))
+                    if (this.VMObject != null)
+                    {
+                        this.ResourceGroupName = this.VMObject.ResourceGroupName;
+                        this.VMName = this.VMObject.Name;
+                    }
+
+                    if (this.InputObject != null)
+                    {
+                        this.ResourceGroupName = this.InputObject.ResourceGroupName;
+                        this.VMName = this.InputObject.VMName;
+                        this.Name = this.InputObject.Name;
+
+                        this.Location = string.IsNullOrEmpty(this.Location) ? this.InputObject.Location : this.Location;
+                        this.TypeHandlerVersion = string.IsNullOrEmpty(this.TypeHandlerVersion) ? this.InputObject.TypeHandlerVersion : this.TypeHandlerVersion;
+                        this.FileUri = (this.FileUri == null || !this.FileUri.Any()) ? this.InputObject.Uri : this.FileUri;
+
+                        string cmdToExe = this.InputObject.CommandToExecute;
```

Comment:
Is there any error checking we should add around these five lines? If not and we are always guaranteed to have the same format you can leave it as is.

Reply:
What kind of format are you talking about?

Reply:
Can you always split InputObject.CommandToExecute into Run and Argument, or could the formatting be different? Or would that fail to create in the first place? If it's not possible to have that incorrect formatting then there's no need to do error checking

Reply:
I changed the code a little bit. But it should always have `-file` since the custom script is a powershell script.